full-stack-tests/presentation/src/frontend/UnifiedSelection.test.ts

```diff
@@ -185,6 +188,172 @@ describe("Unified Selection", () => {
+        };
+        return builder.insertElement(partitionProps);
+      };
+
+      const addModel = (classFullName: string, builder: TestIModelBuilder, partitionId: string) => {
+        const modelProps: ModelProps = {
+          modeledElement: new RelatedElement({ id: partitionId }),
+          classFullName,
+          isPrivate: false,
+        };
+        return builder.insertModel(modelProps);
+      };
+
+      const addSpatialCategory = (builder: TestIModelBuilder, modelId: string, name: string, isPrivate?: boolean) => {
+        const spatialCategoryProps: CategoryProps = {
+          classFullName: "BisCore:SpatialCategory",
+          model: IModel.dictionaryId,
+          code: builder.createCode(modelId, BisCodeSpec.spatialCategory, name),
+          isPrivate,
+        };
+        return builder.insertElement(spatialCategoryProps);
+      };
+
+      const addPhysicalObject = (builder: TestIModelBuilder, modelId: string, categoryId: string, elemCode = Code.createEmpty()) => {
+
+        const physicalObjectProps: PhysicalElementProps = {
+          classFullName: "Generic:PhysicalObject",
+          model: modelId,
+          category: categoryId,
+          code: elemCode,
+        };
+        return builder.insertElement(physicalObjectProps);
+      };
+
+      const objectIdArray = Array<Id64String>();
+      let groupingNodesIModel: IModelConnection;
+      beforeEach(async () => {
+        groupingNodesIModel = await buildTestIModel("GroupByClass", (builder) => {
+          const physicalPartitionId = addPartition("BisCore:PhysicalPartition", builder, "TestDrawingModel");
+          const definitionPartitionId = addPartition("BisCore:DefinitionPartition", builder, "TestDefinitionModel");
+          const physicalModelId = addModel("BisCore:PhysicalModel", builder, physicalPartitionId);
+          const definitionModelId = addModel("BisCore:DefinitionModel", builder, definitionPartitionId);
+          const categoryId = addSpatialCategory(builder, definitionModelId, "Test SpatialCategory");
+          objectIdArray.push(addPhysicalObject(builder, physicalModelId, categoryId, new Code({ value: "code", scope: "0x1", spec: "0x1" })));
+          objectIdArray.push(addPhysicalObject(builder, physicalModelId, categoryId, new Code({ value: "code", scope: "0x1", spec: "0x2" })));
+          objectIdArray.push(addPhysicalObject(builder, physicalModelId, categoryId, new Code({ value: "code2", scope: "0x1", spec: "0x2" })));
+        });
+
+        Presentation.selection.clearSelection("", groupingNodesIModel);
+        handler = new ViewportSelectionHandler({ imodel: groupingNodesIModel });
+      });
+
+      afterEach(async () => {
+        await Presentation.presentation.rulesets().clear();
+        objectIdArray.splice(0);
+        await groupingNodesIModel.close();
+        handler.dispose();
+      });
+
+      it("hilites grouping nodes grouped by class", async () => {
```

Comment:
the test name is interesting... sounds like grouping nodes are grouped by class. IMO, it should be "hilites class grouping node elements". same for other new tests you added.

Reply:
Renamed.